### ISSUE TITLE
citability

### ISSUE THREAD

**Issue by proohit** [WebAssembly/interface-types#133]
Dear authors, thanks for your work on evolving WebAssembly. I am working on my thesis right now, which is all about WebAssembly. I wanted to cite this proposal because it solves problems I encountered during my testings. 

Is there an official way to cite this and possibly other proposals? Thanks in advance.

**Comment by lukewagner** [comment]
That's a good question, thanks for asking.  For my part, I don't know what an official convention would be other than simply including the GitHub URL of this repo (which should stay valid).

**Comment by RossTate** [comment]
Another thing to consider is what should be used as the "author(s)" of the citation. Based on what other communally developed languages do, I would suggest "The WebAssembly Community Group".

**Comment by rossberg** [comment]
> Another thing to consider is what should be used as the "author(s)" of the citation. Based on what other communally developed languages do, I would suggest "The WebAssembly Community Group".

I disagree. Citing the proposal authors' names is good practice, see e.g. [C++ proposals](https://en.wikipedia.org/wiki/C%2B%2B14#References). There is a difference between group decisions and individual authorship. In most cases, the authorship and most of the actual work involved in a proposal is due to few specific people, typically the (co-)champions, sometimes with additional collaborators.

**Comment by RossTate** [comment]
I'm not sure which C++ proposal you're referring to, but sampling through a few I find that their scale is so so so much smaller than the scale of a wasm proposal. (Not to mention that Wikipedia is not particularly great about authors in citations.) And while the champions are conceptually the project lead, "authors" of a work are generally the project contributors, not just leads, and where contributions include not just to the text but also to artifacts that inform, validate, and support the text (e.g. software and experiments). The DOI provided for WASI above lists over 20 authors, which is in line with what is common in research communities where projects are similarly large scale.

To give a non-wasm point of comparison, citing just the champions of a proposal would be like citing just Gilad Bracha for JSR 14, the proposal that added generics to Java. There was a whole paper that led to the first major draft of that design. Then the community tried to put that draft to practice and found it far too limited to be viable, identifying important use cases it failed to satisfy. So then a whole second paper was written to articulate and motivate the second major draft of the design. That then was further iterated by the community until arriving at something that was practically viable and then shipped with Java 5. While Gilad was certainly a huge part in that happening, citing him as sole author would seem to me to underappreciate the many many others involved and their contributions. The convention I have seen in this case is that the citations of the two research papers list their specific authors but citation of the proposal lists the community in some way. That some way was often just Sun Microsystems, as Java is a proprietary language, but other languages have adopted different norms.

So, for citations of a proposal as a whole artifact, I would be in favor of either a broad term, as I suggested above, or a long author list that recognizes the many contributions individuals in the CG make to each proposal, as the DOI above does.

**Comment by proohit** [comment]
I think it's valid to include all participants, i.e. all authors that contributed to the proposal up to now (the time of citing). Maybe in an order similar to how GitHub sorts them to emphasize the leaders of the proposal.

On the other hand, I think it's not common to include all participants. Given I'm writing my thesis, then I myself cite people that supported me to do so. However, if anyone was to cite my thesis, those supporters of my thesis would not be mentioned as authors, even though they contributed.

**Comment by RossTate** [comment]
Heh, yeah, theses are especially weird in that way.

**Comment by rossberg** [comment]
I think citing a GitHub repo is fine. We don't have proper papers for proposals.
